### DISCUSSION TITLE
Support InfluxxDB v2

### DISCUSSION THREAD

**Discussion by nhet** [jomjol/AI-on-the-edge-device#1962] [Ideas] [answered]
Hallöchen und schon mal besten Dank für alles was ihr bisher geschafft habt.
Ich schreibe meine Messwerte direkt in eine Influxx DB. Dies musste ja eine 1.8er-Version sein. Leider gibt es das ein oder andere dokumentierte Probleme in der 1.8er-Version, sodass ich keine Summen, meiner Werte, über einen Zeitraum bilden kann. In dem aktuellen 2.1er soll das Problem nicht auftreten.
Wann plant Ihr die Version Influxx-Version 2 einzubinden?

Danke und LGe

**Comment by jomjol** [comment]
Offen gesprochen gar nicht. Die InfluxDB-Unterstützung wurde von einem User implementiert und nicht aus dem Kernteam. Da das InfluxDB 2.0 Interface komplett anders ist (soweit ich es verstehe), ist schlicht das Know-How erstmal nicht da und der Aufwand recht groß.
Ich selber gehe den Umweg über einen MQTT-Broker, über den ich dann mittels iobroker in die InfluxDB schreibe und dort geht inzwischen auch 2.0.

**Comment by jomjol** [comment]
Habe mir das ganze mal angeschaut und so kompliziert scheint mir der Transfer auf InfluxDBv2 nicht. Bin gerade dabei, etwas zu schreiben und testen. Dauert noch ein bisschen, müsste aber gehen.

Du kannst es hier verfolgen: https://github.com/jomjol/AI-on-the-edge-device/tree/InfluxDB-V2

Ich würde aber nicht zu einer Installation raten, bevor es nicht im Rolling drin ist :-).

**Reply by nhet** [reply]
Hey, 
mega 😍. Bin gespannt.
Danke.

**Comment by jomjol** [comment, marked as answer]
Ich habe jetzt mal eine erste Version, die zumindest bei mir läuft: https://github.com/jomjol/AI-on-the-edge-device/actions/runs/4128260023

Graphische Config habe ich auch schonmal integriert:
![grafik](https://user-images.githubusercontent.com/30766535/217648017-60421cfa-3b8b-4e0b-93e5-ceb93ab40a3b.png)



Die Authentifizierung geht aber nur über einen Token.
Du kannst jetzt in der InfluxDBv2 sogar den Feldnamen manuell eingeben, wenn es notwendig ist. Ist noch eine Beta, aber bei mir läuft es jetzt erstmal ohne Problem. Wenn du also auch testen willst, gerne auf eigenen Gefahr :-)

**Reply by nhet** [reply]
Hey cool,
werd' ich gleich mal testen. Kann dir dann Rückmeldung geben.

**Reply by nhet** [reply]
Auf den ersten Blick und die ersten eingegangenen Werte scheint alles zu klappen.

**Comment by jomjol** [comment]
Ist jetzt im Rolling

**Comment by 8chM** [comment]
Was ist jetzt eigentlich daraus geworden? Wird es jetzt in den neuen Releases implementiert?

**Reply by caco3** [reply]
es ist in release 15.0.3 bereits drinn

**Reply by 8chM** [reply]
Danke dir, freu mich aufs testen.

**Reply by 8chM** [reply]
Bei Home Assistant wird die Version 4.5 verwendet. Kann man diese auch verwenden?

**Reply by caco3** [reply]
was meinst du mit 4.5?

**Comment by friedpa** [comment]
Influxdb.client.core Version 4.5

**Reply by caco3** [reply]
What is the reason you want an update?
Is there something not working?

Looking on the code I think InfluxDB support is implemented without an additional library, so no way/need to update.

**Comment by Surfrobbe** [comment]
Das Speichern direkt in die InfluxDB 2.0 funktioniert bei mir mit der Version 15.3 nicht. Muss ich nicht eigentlich noch irgendwo das Bucket angeben in der er die Daten speichern soll? Ich kann die Daten nur per Mqqt an den iobroker senden und von dort aus in die influxDB schreiben lassen. Allerdings speichert er mir auch die falschen Werte (error) mit ab. Previous Value steht auf true und allow negativ rates auf false.

**Reply by caco3** [reply]
Ich habe es nicht getestet, aber ich vermute, es wird der Parameter `Database` als bucket verwendet, siehe https://github.com/jomjol/AI-on-the-edge-device/blob/1094c8a0a8f0b4eb07d66b1a44c58e72630d0ae1/code/components/jomjol_influxdb/interface_influxdb.cpp#L73


Probier das doch aus und lass es uns wissen, dann können wir das umbenennen/dokumentieren.

**Reply by nhet** [reply]
Moin, ja Database ist das Buket.
![image](https://github.com/jomjol/AI-on-the-edge-device/assets/13017349/ae6f7c33-7295-43af-af4d-384dee9c93b8)

![image](https://github.com/jomjol/AI-on-the-edge-device/assets/13017349/f502831b-2a08-490a-9e8d-995c1b0ace96)

**Comment by Surfrobbe** [comment]
Hat funktioniert. Vielen Dank!

**Reply by caco3** [reply]
Danke für euer Feedback!
Ich denke, wir sollten bei InfluxDb den Parameter `database` in `bucket` umbenennen -> https://github.com/jomjol/AI-on-the-edge-device/issues/2699

**Comment by caco3** [comment]
Ich habe eine neue Firmware gebaut, wo der  Parameter `Database` neu `Basket` heist.
Könnt Ihr mal ausprobieren, ob bei euch noch alles geht?
https://github.com/jomjol/AI-on-the-edge-device/actions/workflows/build.yml?query=branch%3Arename_influxdb2_parameter -> obersten Eintrag, dann runter scollen und das `update` file nehmen.
Beim ersten Aufstarten nach dem Update sollte die Config automatisch umgeschrieben werden.

**Bitte vorher ein Backup der Config machen!**

**Reply by nhet** [reply]
Moin, läuft bei mir nun auf 3 Devices stabil.

**Comment by MaluNoPeleke** [comment]
I bekomme leider keine Daten gesendet, in den Error Logs wird auch nichts angezeigt, was könnte falsch sein?
![image](https://github.com/user-attachments/assets/7f239f06-f7d9-4f4e-ae07-29089412eebf)
![image](https://github.com/user-attachments/assets/2be12429-841b-4b83-9421-3c065a54b75f)
![image](https://github.com/user-attachments/assets/b01d8fb1-cee4-4968-aa27-5b132938fafb)
Wenn ich https://peleke7.public.space in den Browser eingebe, lande ich wie gewünscht in meiner InfluxDB Instanz (bei einem Webhoster)

**Reply by caco3** [reply]
@nhet Mach bitte jeweils eine neue Diskussion auf. Das ist besser als einen alten issue aufzuwärmen.

Ich vermute, dass es an SSL liegen könnte (http**s**), welche von unserer Firmware nicht unterstützt wird.

**Comment by MaluNoPeleke** [comment]
Danke, tatsächlich. Ohne SSL funktioniert es wie gewünscht. Schade, aber gut zu wissen.
Sollte man vielleicht prominenter darauf hinweisen.
